### PR TITLE
webidl: Implement sequences in unions

### PR DESCRIPTION
**Pull request by emilio**
Unblocks #9053

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/9304)

### CODE REVIEW

File: components/script/dom/bindings/codegen/CodegenRust.py [hunk 1]
```diff
@@ -3727,9 +3734,13 @@ def get_match(name):
...
-            assert interfaceObject
-            templateBody = CGList([interfaceObject], "\n")
-            conversions.append(CGIfWrapper("value.get().is_object()", templateBody))
+            assert interfaceObject or arrayObject
+            if interfaceObject:
+                templateBody = CGList([interfaceObject], "\n")
+                conversions.append(CGIfWrapper("value.get().is_object()", templateBody))
+            elif arrayObject:
+                templateBody = CGList([arrayObject], "\n")
+                conversions.append(CGIfWrapper("value.get().is_object()", templateBody))
```
Comment (frewsxcv) [inline components/script/dom/bindings/codegen/CodegenRust.py]: This line can be removed from both branches of the conditional since it's the same

Comment (frewsxcv) [inline components/script/dom/bindings/codegen/CodegenRust.py]: Unless you did that thinking other object types might get added to the conditional

Comment (ecoal95) [inline components/script/dom/bindings/codegen/CodegenRust.py]: Yeah eventually other types might be added.

It seems to me that that line would be necessary in all of them though, so I'll take it out of the conditional.

**Comment by nox** [comment]
What happens if the type is `DOMString or sequence<Blob>`?

-S-awaiting-review +S-awaiting-answer 

---

Reviewed 2 of 3 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/9304#-:-K83LQHq0MDfKDjL2rNA:-816280183)*

**Comment by ecoal95** [comment]
Good catch! (it only generated code for the former).

I fixed it, and also implemented `FromJSValConvertible` for `Root`. It's needed to allow conversion from a JSValue to a `Vec<Root<T>>`.

I think the implementation is correct, but since you probably know better than I if root_from_handlevalue suits correctly there.

**Comment by nox** [comment]
@Ms2ger Do we care about the `Vec<Root<T>>` here? Union types are weird anyway so I'm tempted to say no.

**Comment by Ms2ger** [comment]
Not sure.

**Comment by jdm** [comment]
@nox Care about in what sense? We established that `Vec<Root<T>>` isn't a memory-safety issue, at least.

**Comment by nox** [comment]
@bors-servo r+

@jdm Sure, but that is at odds with `sequence<Blob>` which just ends up as `&[&Blob]` in the Rust code.

**Comment by frewsxcv** [comment]
Related to `Vec<Root<T>>`: https://github.com/servo/servo/issues/8737

**Pull request merged**